python/cuml/common/sparsefuncs.py

```diff
@@ -73,6 +73,23 @@ def _map_l2_norm_kernel(dtype):
     return cuda_kernel_factory(map_kernel_str, dtype, "map_l2_norm_kernel")
 
 
+def _map_mul_sparse_array(dtype):
+    """Creates cupy RawKernel for multiplying sparse array with a dense ar """
+
+    map_kernel_str = r'''
+    ({0} *data, {1} *indices, {2} *indptr, {3} *dense_ar, int data_size) {
+
+      int tid = blockDim.x * blockIdx.x + threadIdx.x;
+
+      if(tid >= data_size) return;
+
+      int index = indices[tid];
+      data[tid] =  data[tid] * dense_ar[index];
```

Comment:
Could this be achieved with indexing? It would be one less kernel that needs to be compiled and maintained. 

`data *= dense_ar[indices]`

Reply:
Agreed, cupy indexing will be cleaner and has less maintenance. 

Made the change with commit  https://github.com/rapidsai/cuml/pull/2575/commits/1efcbde48a7077c2522e70587d96fe68d6fee11c . 

The only downside to this is that our memory overhead increases by a small bit but it really should not be prohibitive given that string processing is anyways mostly expensive, we can revisit this if it becomes prohibitive. 

Let me know this PR looks fine now. :-D